tests/test_meta_functions_against_registry.py

```diff
@@ -9,6 +10,159 @@
 )
 
 
+def test_is_registered(metaregistry, registries, max_pools):
+    for i, registry in enumerate(registries):
+        total_pools = registry.pool_count() if max_pools == 0 else max_pools
+        for pool_index in range(total_pools):
+            pool = registry.pool_list(pool_index)
+            print(pool)
+            if pool != brownie.ZERO_ADDRESS:
+                assert metaregistry.is_registered(pool)
+
+
+def test_get_lp_token(metaregistry, registries, max_pools):
+    for i, registry in enumerate(registries):
+        total_pools = (
+            registry.pool_count() if max_pools == 0 else min(max_pools, registry.pool_count())
+        )
+        for pool_index in range(total_pools):
+            pool = registry.pool_list(pool_index)
+
+            # get_lp_token
+            if (
+                i == METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX
+                or i == METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX
+            ):
+                actual_output = registry.get_lp_token(pool)
+            elif i == METAREGISTRY_STABLE_FACTORY_HANDLER_INDEX:
+                actual_output = pool
+            else:
+                actual_output = registry.get_token(pool)
+
+            metaregistry_output = metaregistry.get_lp_token(pool)
+            assert actual_output == metaregistry_output
+
+
+def test_get_virtual_price_from_lp_token(metaregistry, registries, max_pools, curve_api):
+    for i, registry in enumerate(registries):
+        total_pools = (
+            registry.pool_count() if max_pools == 0 else min(max_pools, registry.pool_count())
+        )
+        for pool_index in range(total_pools):
+            pool = registry.pool_list(pool_index)
+            lp_token = metaregistry.get_lp_token(pool)
+
+            # virtual price from metaregistry:
+            metaregistry_revert = False
+            try:
+                metaregistry_output = metaregistry.get_virtual_price_from_lp_token(lp_token)
+            except:
+                metaregistry_revert = True
+
+            # virtual price from underlying registries (or query chain):
+            actual_output_revert = False
+            try:
+                if (
+                    i == METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX
+                    or i == METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX
+                ):
+                    actual_output = registry.get_virtual_price_from_lp_token(lp_token)
+                else:
+                    actual_output = curve_api.get_virtual_price(pool)
+            except:
+                actual_output_revert = True
+
+            if not (metaregistry_revert and actual_output_revert):
+                assert actual_output == metaregistry_output
+            else:
+                # both should revert for consistency:
+                assert metaregistry_revert and actual_output_revert
+
+
+def test_get_underlying_decimals(metaregistry, registries, max_pools, curve_api):
+    def get_coin_decimals_from_pool(pool):
```

Comment:
A bit unusual to include a function within a function, why not put at the top level?